app/components/videos/videos.controller.js

```diff
@@ -0,0 +1,82 @@
+(function () {
+  let VideosController = function (VideosService, StorageService, $mdToast) {
+    this.$onInit = function () {
+      this.getVideos(this.viewSettings);
+    };
+    this.videosList = [];
+    this.viewSettings = {
+      layout: "list",
+      videosCount: 10,
+      sorts: "newest",
+      filters: null,
+      page: 0
+    };
+    this.openSelectVideosNumber = ($mdOpenMenu, e) => $mdOpenMenu(e);
+    this.changeViewSettings = (paramObj) => {
+      Object.assign(this.viewSettings, paramObj);
+      this.getVideos(this.viewSettings);
+    };
+    this.favToggle = (video) => {
+      video.fav = !video.fav;
+      StorageService.updateVideo(video);
+      $mdToast.show(
+        $mdToast.simple()
+          .textContent(`Video ${ video.name } ${ video.fav ? "added to fav" : "removed from fav"} `)
+          .position("top right")
+          .hideDelay(3000)
+      );
+    };
+
+    this.deleteVideo = (video) => {
+      StorageService.deleteVideo(video);
+      this.getVideos(this.viewSettings);
+      $mdToast.show(
+        $mdToast.simple()
+          .textContent(`Video ${ video.name } deleted `)
+          .position("top right")
+          .hideDelay(3000)
+      );
+    };
+
+    this.getVideos = (paramObject, reset) => {
+      if(reset) this.viewSettings.page = 0;
+      this.videosList = VideosService.getVideosList(paramObject);
```

Comment:
to zdecydowanie powinna byc akcja asynchroniczna. prawdopodobnie dzialajaca tak:
```javascript
VideosService.getVideosList(paramObject)
  .then((data) => this.videosList = data);
```

btw. ja bym inaczej nazwal akcje getVideosList. przeciez DTO nazywa sie Videos, po co wiec masz dodawac, ze pobierasz wlasnie liste videos jesli zaznaczyles skad chcesz to pobierac?